### PR TITLE
Clean up tests in gettext.test.js & ignore translations in test files (extract-translatable-strings.js)

### PR DESCRIPTION
**Pull request by lb-**
As per https://github.com/wagtail/wagtail/pull/8747#issuecomment-1294852488 - the existing unit test for the `gettest` util seemed to make no sense.

However, the purpose of this test file was originally created as we found that the `ngettext` was not actually using the Django global util the correct way by passing all the params through to the global. See https://github.com/wagtail/wagtail/pull/8747

Although, in hindsight, this created two further problems, firstly it meant we got useless translations and secondly we ended up mocking the Django globals which was not really needed.

I propose a different approach to just deleting the file - https://github.com/wagtail/wagtail/pull/9537

1. Update the extract text util to ignore test files
2. Update the unit test file to be more clear about what it's testing; a. the 'default' behaviour if the global does not exist and b. that when the globals exist it calls the global with the correct params (without caring about the output of these mocks).

Note: To test the extract translation strings, do the following:

* Load the branch locally
* `cd client/`
* `node extract-translatable-strings.js`
* Expected result: There should be no new string for the `gettext.test.js` file added to `...admin/locale/en/LC_MESSAGES/djangojs.po` (note: There will be a line change for an unrelated translation usage)

### Checklist

- [X] Do the tests still pass?
- [X] Does the code comply with the style guide?

### CODE REVIEW

**Comment by lb-** [comment]
Created as an alternative approach to https://github.com/wagtail/wagtail/pull/9537 (just deleting the test)

**Review by allcaps — APPROVED** [review]

**Comment by lb-** [comment]
Thanks @allcaps  - will merge in.

File: client/src/utils/gettext.test.js [hunk 1]
```diff
@@ -9,68 +9,74 @@ import {
...
+    gettext(STRING);
+    expect(gettextMock).toHaveBeenCalledWith(STRING);
   });
 });
 
 describe('ngettext', () => {
-  beforeEach(() => {
-    window.django = { ngettext: jest.fn((val) => val) };
+  it('should return the first param if Django ngettext is not loaded', () => {
+    expect(ngettext('One bird', 'Many birds', 2)).toEqual('One bird');
```
Comment (gasman) [inline client/src/utils/gettext.test.js:34]: @lb- Why doesn't this return "Many birds"?

Comment (lb-) [reply client/src/utils/gettext.test.js:34]: Because. This is the default behaviour if the global Django util is not loaded. 

That's been the default behaviour from the time that this code was introduced. 

Sorry if it's not clear but this unit test checks that the code has some default behaviour (not an error) if the global is loaded. 


The unit test after checks that if the global is loaded, it gets called. 

Would it help if I just changed the number to 1?

Comment (gasman) [reply client/src/utils/gettext.test.js:34]: Ah yep, makes sense, thanks! I missed the updated description of the test.

Just so I'm clear - the Django util functions being absent (and the use of these fallback functions) isn't something we expect to happen in "real-world" code, but only for the purposes of unit tests, right? I'd be a bit happier if the fallback implementation became `return (count == 1 ? singular : plural)`, so that the behaviour in tests is equivalent to running it for real in an English-language locale.

Comment (lb-) [reply client/src/utils/gettext.test.js:34]: Sounds good. I'll prepare another PR. 

Alternatively we could make it error loudly if the global is not there, it's a bit more work to change it but maybe that is more 'correct'?

We would however need to ensure that all unit tests have a mock of the global loaded (can be done in jest.setup) or a mock for each of the utils.

It's just a bit tricky either way as the Django import adds a global instead of being an npm module import. So we kind of need to think about the low chance of it not working but also handle the import global not existing sometimes like unit tests or maybe Storybook in edge cars.

Comment (gasman) [reply client/src/utils/gettext.test.js:34]: I think the "make it error loudly" approach would probably just make life difficult for us :-) We're best served by having a simple fallback implementation that doesn't have any external dependencies, but is close enough to the "real" behaviour that our tests don't have to be contrived to work around the differences (e.g. being able to assert that the output contains "2 Errors" rather than "2 Error").
